src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.WorkspaceHost.cs

```diff
@@ -45,26 +54,48 @@ public void OnSolutionAdded(SolutionInfo solutionInfo)
                 RegisterPrimarySolutionAsync().Wait();
             }
 
+            private async Task<ReferenceCountedDisposable<Connection>> GetConnectionAsync()
+            {
+                this.AssertIsForeground();
```

Comment:
❔ Does this really need to be on the UI thread? I notice that `TryCreateConnectionAsync` uses `ConfigureAwait(false)` so we don't necessarily *stay* on the UI thread.

:bulb: If something here *does* need to be on the UI thread, why not use the equivalent of this?

```
await SwitchToMainThreadAsync();
```